pkg/lib/registry/registry.go

```diff
@@ -354,3 +368,39 @@ func (r RegistryUpdater) DeprecateFromRegistry(request DeprecateFromRegistryRequ
 
 	return nil
 }
+
+// CheckForBundlePaths verifies presence of a list of bundle paths in the registry.
+func CheckForBundlePaths (querier registry.Query, bundlePaths []string) (found []string, missing []string, err error) {
+	if len(bundlePaths) == 0 {
+		return bundlePaths, nil, nil
+	}
+
+	registryBundles, err := querier.ListBundles(context.TODO())
+	if err != nil {
+		return bundlePaths, nil, err
+	}
+
+	if len(registryBundles) == 0 {
+		return nil, bundlePaths, nil
+	}
+
+	registryBundlePaths := make(map[string]struct{})
+	for _, b := range registryBundles {
+		registryBundlePaths[b.BundlePath] = struct{}{}
+	}
+
+	for _, b := range bundlePaths {
+		if _, ok := registryBundlePaths[b]; ok {
+			if found == nil {
+				found = make([]string, 0)
+			}
+			found = append(found, b)
+			continue
+		}
+		if missing == nil {
+			missing = make([]string, 0)
+		}
+		missing = append(missing, b)
```

Comment:
Am I missing something here? It looks like missing was never declared before referencing it here.

Reply:
missing was declared in the function signature, it looked easier to understand CheckForBundlePaths that way